File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> op(numRows);
        for(int i=0;i<numRows;i++){
            op[i].resize(i+1);
            op[i][0] =op[i][i]=1;
             for(int j=1;j<i;j++)
              {
                 op[i][j] = op[i-1][j-1] + op[i-1][j];
              }
        }
        return op;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, n: int) -> List[List[int]]:
        ans = []
        for i in range(n):
            t = [1] * (i + 1 )
            for j in range(i):
                if j == 0 or j == i + 1:
                    continue
                else:
                    t[j] = ans[i-1][j-1] + ans[i-1][j]
            ans.append(t)
        return ans